scheduler.go

```diff
@@ -213,9 +213,11 @@ func (s *Scheduler) RemoveByReference(j *Job) {
 }
 
 func (s *Scheduler) removeByCondition(shouldRemove func(*Job) bool) {
-	for i, job := range s.jobs {
-		if shouldRemove(job) {
+	for i := 0; i < len(s.jobs); {
+		if shouldRemove(s.jobs[i]) {
```

Comment:
Will `s.jobs[i]` not panic?
Because we are iterating over `len(s.jobs)` and removing jobs in the same loop.

Maybe we should maintain a separate array of jobs that are not removed and finally assign this new array to `s.jobs`:

```go
retainedJobs := []<job-type>
if !shouldRemove(s.jobs[i]) {
    retainedJobs = append(retainedJobs, s.jobs[i])
}
s.jobs = retainedJobs
```
This would perhaps be safer.

Reply:
What do you guys think?
@JohnRoesler @Streppel

Reply:
So, we're not actually removing any job but instead keeping the job that shouldn't be removed and assigning the new slice to `s.jobs`?

Reply:
Yes.

Reply:
Okay, should I wait for other reviewers or change it right now?

Reply:
Let's hear what the others have to say

Reply:
Thank you @ridwanakf for your contribution and @arjunmahishi for reviewing and pinging us out! 

I think @arjunmahishi  comment is absolutely valid and a good enhancement for the project. We should avoid mutate slices while iterating over them. Can you make this change @ridwanakf ?

Reply:
Sure, I'll work on that. btw, should I just revert back the loop using `range` since I think it's rather cleaner and also not really the problem anymore since we just keeping track of new slice?

Reply:
Feel free to do so :)

Reply:
Nevermind, I think it doesn't matter since we will be assigning `s.jobs` to new slice anyway. CMIIW

Reply:
I've made the changes, please kindly review it again.